kong/plugins/ip-restriction/handler.lua

```diff
@@ -29,6 +33,31 @@ do
 end
 
 
+local is_http_subsystem = ngx.config.subsystem == "http"
+
+
+local do_exit
+if is_http_subsystem then
+  do_exit = function(status, message)
+    return kong.response.error(status, message)
+  end
+
+else
+  do_exit = function(status, message)
+    local tcpsock, err = ngx_req.socket(true)
+    if err then
+      error(err)
+    end
+
+    tcpsock:send(cjson_encode({
+      status  = status,
+      message = message
```

Comment:
Why send a 'status' property for TCP/UDP connections? HTTP status is an L7 concept and we should not be sending that to a client who is using TCP/UDP.

Reply:
I understand your point.  What type of response would you recommend?

Reply:
@hbagdi Any suggestion on the response format and content?

Reply:
`{"message":"rate-limit exceeded"}` would make more sense.

Reply:
So would the following be acceptable?

```lua
    tcpsock:send(cjson_encode({
      message = message
    }))
```

Reply:
Status removed